lib/datadog/appsec/event.rb

```diff
@@ -137,6 +137,24 @@ def build_service_entry_tags(event_group)
         end
         # rubocop:enable Metrics/MethodLength
 
+        def add_event_tags(scope, waf_result)
+          if scope.service_entry_span
+            scope.service_entry_span.set_tag('appsec.blocked', 'true') if waf_result.actions.include?('block')
+            scope.service_entry_span.set_tag('appsec.event', 'true')
+          end
+
+          # Propagate to downstream services the information that the current distributed trace is
+          # containing at least one ASM security event
+          return unless scope.trace
+
+          scope.trace.keep!
+          scope.trace.set_tag(
+            Datadog::Tracing::Metadata::Ext::Distributed::TAG_DECISION_MAKER,
+            Datadog::Tracing::Sampling::Ext::Decision::ASM
+          )
+          scope.trace.set_tag(Datadog::AppSec::Ext::TAG_APPSEC_EVENT, '1')
```

Comment:
Do we want to execute this code only if the standalone appsec is enabled?

Reply:
> Do we want to execute this code only if the standalone appsec is enabled?

We want to execute this code each time there is an Appsec event, it is not specific to Appsec standalone

I wrote this method to factorise `watch_xxx` methods in each integration's Gateway Watcher so that it would comply to our Rubocop spec.

> and I would actually move this code to another function

Would you move it to another method because it is acting on `scope.trace` and not `scope.service_entry_span` anymore ?

Reply:
yes, I think it does a bit more then setting tags - if I understood correctly it's more about making sure that the trace is preserved?